templates/repo/diff/stats.tmpl

```diff
@@ -1,4 +1,5 @@
 {{Eval .file.Addition "+" .file.Deletion}}
 <span class="diff-stats-bar gt-mx-3" data-tooltip-content="{{.root.locale.Tr "repo.diff.stats_desc_file" (Eval .file.Addition "+" .file.Deletion) .file.Addition .file.Deletion | Str2html}}">
-	<div class="diff-stats-add-bar" style="width: {{DiffStatsWidth .file.Addition .file.Deletion}}%"></div>
+	{{/* if the denominator is zero, then the float result is "width: NaNpx", as before, it just works */}}
+	<div class="diff-stats-add-bar" style="width: {{Eval 100 "*" .file.Addition "/" "(" .file.Addition "+" .file.Deletion "+" 0.0 ")"}}%"></div>
```

Comment:
What is the purpose of `+ 0.0`?

Reply:
Likely to ensure it's a float

Reply:
Depending on the operands it might. 
https://go.dev/play/p/XZrH-1pxLMk

Reply:
Go math is weird. Float return value for division would have been more sane with rounding being another function call. But I guess the go authors have a bias towards integers so go is unsuitable for scientific calculation 😆.

Reply:
Another purpose, in Golang:

```go
	a := 1
	fmt.Println(5 / (a - 1))
```

There will be a panic: `panic: runtime error: integer divide by zero`.

But if it does float math: `fmt.Println(5 / (float64(a) - 1))`, it gets `+Inf`, `-Inf` or `NaN`.

At the moment, the `Eval` just follows the Golang's standard strictly, and that's what `DiffStatsWidth` / `percentage` help functions did: use float, return NaN.

----

If we prefer to avoid the "integer divide by zero" panic for integers, we can add our rule, eg: use float `NaN` instead.


However, I didn't do too much customization because I think the "percentage" behavior is not clear for UI at the moment, for example, "0/0" could mean "0%", "50%" or "100%" on the UI, it depends its meaning and context.